### ISSUE TITLE
Buffering / Stream support

### ISSUE THREAD

**Issue by stavros-zavrakas** [zmxv/react-native-sound#353]
Is there a plan to support buffering / stream support?

I've tried to use the sound from the network but I wasn't successful. There is this module (https://github.com/tlenclos/react-native-audio-streaming) that is not really maintained anymore but it is able to stream from the network. Do you think that this can be supported at some point in react-native-sound?

**Comment by mateioprea** [comment]
@stavros-zavrakas  use react-native-video for streaming for now

**Comment by johncblandii** [comment]
That's a pretty sucky response to what is clearly a needed feature. Why `wontfix` vs, at a minimum, putting it on the backlog?

**Comment by trepidity** [comment]
OK, sorry. I didn't explain because I felt that the issue had been discussed in other topics.

The deal is that this library doesn't use a library that supports streaming #292. We would have to rewrite and use the correct library. I don't know the amount of work required, but I bet it's not trivial. If someone wants to tackle that, great, but I'm not aware of anyone so far.

If someone wants to submit a PR with the correct AVPLayer, I'm all game for that.

**Comment by ohtangza** [comment]
@johncblandii @trepidity I agree that supporting streaming is not easy because it requires lots of work in native side (almost equivalent to re-writing this library). Since this library is used by production apps including our team's, I guess we need to focus on stabilization of the existing functionality. Currently, too many issues open, so this library is getting messy.

**Comment by johncblandii** [comment]
@trepidity that's clearer. If I get enough free time, I'll push a PR.

@ohtangza feel free to focus on stabilization. Just when network playback becomes a feature the natural next step is streaming audio (either via progressive download or otherwise). Labeling it as help wanted is a great step forward to getting the help from those of us in the community capable of such a rewrite.

**Comment by ohtangza** [comment]
@johncblandii Thanks for your prompt input on this. As you said, we'd like to make it clear somewhere in docs about the directions. Please understand that this is not a judgment on whether the feature is important or not. It's just not a trivial work because we need to change the core part of this library as @trepidity commented.  If you submit a pull request to implement streaming, my team would be happy to review it because we could enhance our user's experience to a certain degree :) 

In the meantime, `react-native-audio-toolkit` could be your second choice because it supports streaming very well in ios and android when we prototyped with this library earlier.

**Comment by johncblandii** [comment]
No problem, @ohtangza. I get where you guys are coming from for sure.

**Comment by eis** [comment]
Regarding `react-native-audio-toolkit`, it seems they have a problem of not being maintained either (https://github.com/futurice/react-native-audio-toolkit/issues/83) and they don't support local playback in ios, which is kinda important ([SOURCES.md](https://github.com/futurice/react-native-audio-toolkit/blob/master/docs/SOURCES.md))

**Comment by dmathewwws** [comment]
If you are looking for other options, there is also https://github.com/react-native-kit/react-native-track-player which supports streaming

**Comment by RamanSB** [comment]
> If you are looking for other options, there is also https://github.com/react-native-kit/react-native-track-player which supports streaming

Curious to know how this performed for everybody?
